src/util.ts

```diff
@@ -494,35 +496,81 @@ export function selectDefFrom4byteABI (abiData: any[], selector: string) {
   }
 }
 
-async function fetchSupportedChainData (address: string, supportedChain: number) {
-  const url = buildUrlForSupportedChainAndAddress({ address, supportedChain })
-  return superagent
-    .get(url)
-    .then(async res => {
-      if (res && res.body && res.body.result) {
-        return JSON.parse(res.body.result)
+export async function fetchWithTimeout (url, options) {
+  const { timeout } = options;
```

Comment:
Probably want to set a default, no?

Reply:
There's a default set in the `request` function. I suppose it _does_ make sense to set another, shorter one here if we use it elsewhere.